kin-core/src/main/java/kin/core/AccountInfoRetriever.java

```diff
@@ -2,27 +2,40 @@
 
 
 import android.support.annotation.NonNull;
+import android.text.TextUtils;
+
 import java.io.IOException;
 import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
 import kin.core.Environment.KinAsset;
 import kin.core.exception.AccountNotActivatedException;
 import kin.core.exception.AccountNotFoundException;
 import kin.core.exception.OperationFailedException;
 import org.stellar.sdk.KeyPair;
 import org.stellar.sdk.Server;
+import org.stellar.sdk.requests.RequestBuilder;
+import org.stellar.sdk.requests.TransactionsRequestBuilder;
 import org.stellar.sdk.responses.AccountResponse;
 import org.stellar.sdk.responses.HttpResponseException;
+import org.stellar.sdk.responses.Page;
+import org.stellar.sdk.responses.TransactionResponse;
 
 class AccountInfoRetriever {
 
     private final Server server;
     private final KinAsset kinAsset;
+    private BlockchainEvents blockchainEvents;
```

Comment:
can you extract the `TransactionResponse` -> `PaymentInfo` logic to a common class? it's not entirely clean to have ` AccountInfoRetriever` depends on unrelated `BlockchainEvents` class